api/src/main/java/org/pmiops/workbench/workspaces/WorkspacesController.java

```diff
@@ -460,9 +545,18 @@ private void setDbWorkspaceFields(
 
     dbWorkspace.setBillingMigrationStatusEnum(BillingMigrationStatus.NEW);
 
-    DbWorkspace savedDbWorkspace =
-        workspaceService.saveAndCloneCohortsConceptSetsAndDataSets(fromWorkspace, dbWorkspace);
+    updateWorkspaceBillingAccount(dbWorkspace, body.getWorkspace().getBillingAccountName());
+
+    try {
+      dbWorkspace =
+          workspaceService.saveAndCloneCohortsConceptSetsAndDataSets(fromWorkspace, dbWorkspace);
+    } catch (Exception e) {
+      // Tell Google to set the billing account back to the free tier if our clone fails
+      updateWorkspaceBillingAccount(dbWorkspace, workbenchConfigProvider.get().billing.accountId);
+      throw e;
+    }
 
+    // TODO: File a bug about this. Can create a workspace
```

Comment:
Could you go into more detail here?

Reply:
oops. Yeah, I expanded my comment. I'm not quite sure if it warrants a ticket so I think I'm going to leave it at that